referenceframe/frame_system.go

```diff
@@ -586,8 +586,14 @@ func createFramesFromPart(part *FrameSystemPart) (Frame, Frame, error) {
 		if err != nil {
 			return nil, nil, err
 		}
+		pose, err := staticOriginFrame.Transform([]Input{})
+		if err != nil {
+			return nil, nil, err
+		}
 		if len(offsetGeom.Geometries()) > 0 {
-			modelFrame = &noGeometryFrame{modelFrame}
+			if modelFrame, err = NewStaticFrame(modelFrame.Name(), pose); err != nil {
```

Comment:
This is dangerous; if something happens such that this becomes a `:=` in the future, things will invisibly break.

Instead of setting `modelFrame` directly here, create a new variable, and if err == nil, assign `modelFrame` to the new thing.

Reply:
I don't understand.  It won't even compile if it changes to a `:=`

Reply:
No as-is, no, but it's ripe for a future change to cause a scoping issue that would be hard to notice. Plus it's not good style to be overwriting variables before you've done an error check. In an area of code where an error wouldn't cause us to return nil, this paradigm causes problems.

Instead:

```
// If there are offset geometries, they should replace the static geometries, so the static frame is recreated with no geoms
noGeomFrame, err := NewStaticFrame(modelFrame.Name(), pose)
 if err != nil {
  return nil, nil, err
}
modelFrame = noGeomFrame
```